upload/common/checksum.py

```diff
@@ -37,7 +37,8 @@ def checksum(self, report_progress=False):
         else:
             progress_callback = None
 
-        return self._compute_checksums(progress_callback=progress_callback)
+        self.checksums = self._compute_checksums(progress_callback=progress_callback)
```

Comment:
I don't see this instance variable used anywhere.  Not sure there's real value in assigning to a variable before immediately returning it.  If you're trying to document what the function is returning, a local variable would be fine for that.